src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +451,338 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+                                new XElement(snippetNamespace + "Title", new XText(methodName)),
+                                new XElement(snippetNamespace + "Author", "Microsoft"),
+                                new XElement(snippetNamespace + "Description"),
+                                new XElement(snippetNamespace + "HelpUrl"),
+                                new XElement(snippetNamespace + "Shortcut", methodName)),
+                            new XElement(
+                                snippetNamespace + "Snippet",
+                                new XElement(
+                                    snippetNamespace + "Declarations",
+                                    new XElement(
+                                        snippetNamespace + "Literal",
+                                        new XAttribute(snippetNamespace + "Editable", "true"),
+                                        new XElement(snippetNamespace + "ID", new XText("placeholder")),
+                                        new XElement(snippetNamespace + "ToolTip", new XText("")),
+                                        new XElement(snippetNamespace + "Default", new XText("")),
+                                        new XElement(snippetNamespace + "Function"))),
+                                new XElement(
+                                    snippetNamespace + "Code",
+                                    new XAttribute(snippetNamespace + "Language", "csharp"),
+                                    new XCData(template))))));
+
+                var doc = new DOMDocumentClass();
+                if (doc.loadXML(snippet.ToString(SaveOptions.OmitDuplicateNamespaces)))
+                {
+                    var methodSymbols = symbols.OfType<IMethodSymbol>().ToImmutableArray();
+                    _symbols = methodSymbols;
+                    _symbol = null;
+
+                    if (expansion.InsertSpecificExpansion(doc, textSpan, this, LanguageServiceGuid, pszRelativePath: null, out ExpansionSession) == VSConstants.S_OK)
+                    {
+                        Debug.Assert(_symbols == methodSymbols);
+                        Debug.Assert(_symbol == null);
+
+                        // Trigger signature help after starting the snippet session
+                        var workspace = (VisualStudioWorkspace)SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges()!.Project.Solution.Workspace;
+                        var exportProvider = (IMefHostExportProvider)workspace.Services.HostServices;
+                        var editorCommandHandlerServiceFactory = exportProvider.GetExports<IEditorCommandHandlerServiceFactory>().Single().Value;
+                        var editorCommandHandlerService = editorCommandHandlerServiceFactory.GetService(TextView, SubjectBuffer);
+                        editorCommandHandlerService.Execute((view, buffer) => new InvokeSignatureHelpCommandArgs(view, buffer), nextCommandHandler: null);
```

Comment:
consider doc'ing why `null` is hte right thing to pass here for `nextCommandHandler`

Reply:
I have no idea why it's necessary to do this. I expected to be using `ISignatureHelpBroker` to trigger signature help through the API but it didn't work (unknown reason). Invoking like this worked. Will add a TODO comment to code to figure out why this works.

Reply:
➡️ Tracking this under https://github.com/dotnet/roslyn/pull/49916#discussion_r544818461